File: spring-boot/src/main/java/zencode/sb/portfolio/security/ApplicationUserRepository.java
package zencode.sb.portfolio.security;

import org.springframework.stereotype.Repository;

import javax.annotation.PostConstruct;
import java.util.HashMap;
import java.util.Map;

/**
 * @author Koert Zeilstra
 */
@Repository
public class ApplicationUserRepository {

  private Map<String, String> passwords = new HashMap<>();

  @PostConstruct
  public void initialize() {
    this.passwords.put("guest", "$2a$10$sekw7hnxOLRfOvJYBnt12uc3GhQl9Rk.NKqPjLpbyYy1gfcUyCm0i");
    this.passwords.put("koert", "$2a$10$bTJngWsw6WIwqxjZXj/j7eUBppFfFyt1STE3Q7gO6mO0YS.vbfpsq");
  }

  public ApplicationUser findByUsername(String userName) {
    ApplicationUser applicationUser = null;
    if (this.passwords.containsKey(userName)) {
      applicationUser = new ApplicationUser();
      applicationUser.setUserName(userName);
      applicationUser.setPassword(this.passwords.get(userName));
    }
    return applicationUser;
  }
}

File: stock-sb/src/main/kotlin/nl/zencode/port/stock/security/ApplicationUserRepository.kt
package nl.zencode.port.stock.security

import org.springframework.stereotype.Repository
import java.util.*
import javax.annotation.PostConstruct

/**
 * @author Koert Zeilstra
 */
@Repository
class ApplicationUserRepository {
  private val passwords: MutableMap<String, String?> = HashMap()
  @PostConstruct
  fun initialize() {
    passwords["guest"] = "$2a$10\$sekw7hnxOLRfOvJYBnt12uc3GhQl9Rk.NKqPjLpbyYy1gfcUyCm0i"
    passwords["koert"] = "$2a$10\$bTJngWsw6WIwqxjZXj/j7eUBppFfFyt1STE3Q7gO6mO0YS.vbfpsq"
  }

  fun findByUsername(userName: String): ApplicationUser? {
    var applicationUser: ApplicationUser? = null
    if (passwords.containsKey(userName)) {
      applicationUser = ApplicationUser(userName, passwords[userName])
    }
    return applicationUser
  }
}